sw/device/tests/dif/dif_uart_test_mocked.cc

```diff
@@ -0,0 +1,734 @@
+// Copyright lowRISC contributors.
```

Comment:
Google C++ requires that this file live adjacent to the header/TU pair it tests. Please move this file into `sw/device/lib/dif`.

Reply:
We had this discussion earlier with @lenary and @alphan included. @lenary and myself agree that it is weird to keep CI only code together with the production code in lib.

Reply:
I strongly disagree that that is weird. Tests are not CI-only code; they act as a kind of documentation that should be easy to find for anyone using those DIFs, and, as such, should live alongside the code that they test.

Reply:
Well I think more to the point that something that will never be compiled and ran for the target lives together with something that always is compiled and is ran on the target is a bit confusing.

Reply:
I chatted with Garret about this. Normally, I would still push to have everything in the same directory, but the real point is that it needs to be easy to find unit tests _and_ integration tests, and those can't all live in the same directory, so we're going to need a more disciplined story.

I think that, for now, you should just do whatever, or follow the existing convention of the two existing in-tree unit tests. Placement of tests is something we need to decide and document, rather than just doing whatever.

Also, me and Garret decided that the "_test" prefix is potentially inappropriate, given that it's a very overloaded term, so we might want to go with something like "_unittest" (the fact that the tests use mocks is pretty irrelevant and should not be in the name: print_test.cc, for example, does not use mocks, but is also a unit test).

Reply:
I'm happy for the file suffix to be `_unittest` (rather than `_test_mocked`).